From b541591b416ec96fd8d4a33da0b9ac8c792bfd56 Mon Sep 17 00:00:00 2001
From: raysenlau <raysen@live.cn>
Date: Tue, 11 Aug 2020 12:58:09 +0800
Subject: [PATCH 1/2] FingerprintService: hide InDisplayFingerprintView on
 Error

All defined errors are not recoverable. Thus, InDisplayFingerprintView
should not continue to be shown.

Change-Id: I05e5a82068e089f63a1cf06e5b1d743e82fc8a50
Signed-off-by: Jesse Chan <jc@lineageos.org>
---
 .../biometrics/fingerprint/FingerprintService.java       | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/services/core/java/com/android/server/biometrics/fingerprint/FingerprintService.java b/services/core/java/com/android/server/biometrics/fingerprint/FingerprintService.java
index da5baad4b..a14806457 100644
--- a/services/core/java/com/android/server/biometrics/fingerprint/FingerprintService.java
+++ b/services/core/java/com/android/server/biometrics/fingerprint/FingerprintService.java
@@ -675,6 +675,15 @@ public class FingerprintService extends BiometricServiceBase {
                         mCurrentUserId = UserHandle.USER_NULL;
                     }
                 }
+                if (mHasFod && error != BiometricConstants.BIOMETRIC_SUCCESS
+                        && error != BiometricConstants.BIOMETRIC_ERROR_VENDOR
+                        && error < BiometricConstants.BIOMETRIC_ERROR_VENDOR_BASE) {
+                    try {
+                        mStatusBarService.hideInDisplayFingerprintView();
+                    } catch (RemoteException e) {
+                        Slog.e(TAG, "hideInDisplayFingerprintView failed", e);
+                    }
+                }
             });
         }
 
-- 
2.27.0

